python/cuspatial/cuspatial/core/spatial/distance.py

```diff
@@ -91,9 +91,10 @@ def directed_hausdorff_distance(xs, ys, space_offsets):
         ys,
         as_column(space_offsets, dtype="uint32"),
     )
-    result = result.data_array_view
-    result = result.reshape(num_spaces, num_spaces)
-    return DataFrame(result)
+    with cudf.core.buffer.acquire_spill_lock():
+        result = result.data_array_view(mode="write")
```

Comment:
Why does this require `write` access to the view? I don't think the below changes the buffer.

Reply:
Right, we are also performing a reshape too. Make sense to do a read-only view here.